src/stores/subgraphStore.ts

```diff
@@ -0,0 +1,254 @@
+    )
+  }
+
+  override async save(): Promise<UserFile> {
+    this.validateSubgraph()
+    if (!this.hasPromptedSave) {
+      //Swap to saveAs?
+      const newname = await useDialogService().prompt({
+        title: t('subgraphStore.saveBlueprint'),
+        message: t('subgraphStore.blueprintName') + ':',
+        defaultValue: this.filename
+      })
+      if (!newname) return this
+      this.hasPromptedSave = true
+      this.updatePath(SubgraphBlueprint.basePath + newname + '.json')
+    }
+    useSubgraphStore().updateDef(this as LoadedComfyWorkflow)
+    return super.save()
+  }
+
+  //TODO: reconsider semantics here with promptonce
+  //Should update current name, but keep true?
+  override async saveAs(path: string) {
+    this.validateSubgraph()
+    this.hasPromptedSave = true
+    useSubgraphStore().updateDef(this as LoadedComfyWorkflow)
+    return await super.saveAs(path)
+  }
+}
+
+const subgraphCache: Record<string, LoadedComfyWorkflow> = {}
+
+export const useSubgraphStore = defineStore('subgraph', () => {
+  const typePrefix = 'SubgraphBlueprint'
+  const subgraphDefCache = ref<Map<string, ComfyNodeDefImpl>>(new Map())
+  const canvasStore = useCanvasStore()
+  const dialogService = useDialogService()
+  const subgraphBlueprints = computed(() => [
+    ...subgraphDefCache.value.values()
+  ])
+  //TODO: On initialization, send background request to resolve definitions.
+  //TODO: Find someway to actually signal that information is not ready for synchronous queries
+  async function fetchSubgraphs() {
+    //TODO: lighten this call?
+    const res = (
+      await api.listUserDataFullInfo(SubgraphBlueprint.basePath)
+    ).filter((f) => f.path.endsWith('.json'))
+    await Promise.all(
+      res.map(async (f) => {
+        const name = f.path.slice(0, -5)
+        f.path = SubgraphBlueprint.basePath + f.path
+        const bp = await new SubgraphBlueprint(f).load()
+        useWorkflowStore().attachWorkflow(bp)
+        const nodeDef = convertToNodeDef(bp)
+
+        subgraphDefCache.value.set(name, nodeDef)
+        subgraphCache[name] = bp
+      })
+    )
+  }
+  function convertToNodeDef(workflow: LoadedComfyWorkflow): ComfyNodeDefImpl {
+    const name = workflow.filename
+    const subgraphNode = workflow.changeTracker.initialState.nodes[0]
+    if (!subgraphNode) throw new Error('Invalid Subgraph Blueprint')
+    const uuid = subgraphNode.type
+    if (!uuid) throw new Error('')
+    subgraphNode.inputs ??= []
+    subgraphNode.outputs ??= []
+    //NOTE: Types are cast to string. This is only used for input coloring on previews
+    const inputs = Object.fromEntries(
+      subgraphNode.inputs.map((i) => [
+        i.name,
+        ['' + i.type, undefined] satisfies InputSpec
+      ])
+    )
+    const nodedefv1: ComfyNodeDefV1 = {
+      input: { required: inputs },
+      output: subgraphNode.outputs.map((o) => '' + o.type),
```

Comment:
Same here, unless there's a style guide rule that string casts are done by concatenating `''`